src/components/CookieNotice/CookieNotice.tsx

```diff
@@ -0,0 +1,132 @@
+
+  const [showDialog, setShowDialog] = React.useState(false);
+  const handleShowDialog = () => {
+    setShowDialog(!showDialog);
+  };
+
+  const shouldShowCookieNotice = () => !localStorage.getItem(scrumlrCookieName); // function: checks if scrumlrCookieName is present in localStorage.
+  // If not, shouldShowCookieNotice becomes true.
+  /**
+   * Saves cookie consent to storage.
+   * @param value: true or false, depending on given consent for cookies
+   * @param key: scrumlrCookieName, i.e. scrumlr_consent
+   */
+  const saveToStorage = (key: string, value: string) => {
+    localStorage.setItem(key, value);
+    document.cookie = `${key}=${value};`;
+  };
+
+  /**
+   * Sets cookie consent value i.e. scrumlrCookieName true.
+   */
+  const acceptFunction = () => {
+    if (shouldShowCookieNotice()) {
+      saveToStorage(scrumlrCookieName, "true");
+      setShowCookieNotice(!showCookieNotice);
+      store.dispatch(ActionFactory.addCookie(scrumlrCookieName, true));
+    }
+  };
+
+  /**
+   * Sets cookie consent value i.e. scrumlrCookieName false.
+   */
+  const declineFunction = () => {
+    if (shouldShowCookieNotice()) {
+      saveToStorage(scrumlrCookieName, "false");
+      setShowCookieNotice(!showCookieNotice);
+      store.dispatch(ActionFactory.addCookie(scrumlrCookieName, false));
+    }
+  };
+
+  const openPolicy = () => {
+    setShowDialog(true);
+  };
+
+  if (!shouldShowCookieNotice()) return null; // prevents cookie notice being shown if scrumlrCookieName is set.
+
+  // cookie notice rendering
+  return (
+    <div className={showCookieNotice ? "cookie-notice--show" : "cookie-notice--hidden"}>
+      <Backdrop
+        className="cookie-notice__backdrop"
+        invisible
+        open={open}
+        onClick={() => {
+          setOpen(false);
+          acceptFunction();
+        }}
+      />
+
+      <div className="cookie-notice__header">
+        <h3>Cookies</h3>
+      </div>
+
+      <div className="cookie-notice__body">
+        <p>
+          We and selected partners use cookies or similar technologies as specified in the cookie policy. You can consent to the use of such technologies by closing this notice, by
+          interacting with any link or button outside of this notice or by continuing to browse otherwise.
+        </p>
+      </div>
+      <div className="cookie-notice__buttons">
+        <button className="cookie-notice__button-cookie-policy" type="button" onClick={openPolicy}>
+          Learn more about our Cookie Policy
+        </button>
+
+        <button className="cookie-notice__button-decline" type="button" onClick={declineFunction}>
+          Decline
+        </button>
+
+        <button className="cookie-notice__button-accept" type="button" color="primary" onClick={acceptFunction}>
+          Accept
+        </button>
+      </div>
+      <CookiePolicy scrumlrCookieName={scrumlrCookieName} acceptFunction={acceptFunction} onClose={handleShowDialog} show={showDialog} />
```

Comment:
If you export the `COOKIE_CONSTENT_NAME` you can just access it as a variable instead of passing it as a prop for the CookiePolicy component here. _qed._ :D